### ISSUE TITLE
linter: no-unused-vars incorrectly report imported types used in tsdoc

### ISSUE THREAD

**Issue by posva** [oxc-project/oxc#11639]
### What version of Oxlint are you using?

1.1.0

### What command did you run?

`oxlint`

### What does your `.oxlintrc.json` config file look like?

```jsonc
// noconfig
```

### What happened?

It fails with this code:

```ts
import type { someFunction } from './some-file'

/**
 * Description
 *
 * @see {@link someFunction}
 */
export function myFn() {}
```
